src/rpc/blockchain.cpp

```diff
@@ -1646,34 +1647,141 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
     ret.pushKV("mempoolminfee", ValueFromAmount(std::max(pool.GetMinFee(maxmempool), ::minRelayTxFee).GetFeePerK()));
     ret.pushKV("minrelaytxfee", ValueFromAmount(::minRelayTxFee.GetFeePerK()));
     ret.pushKV("unbroadcastcount", uint64_t{pool.GetUnbroadcastTxs().size()});
+
+    if (feeLimits) {
+        const MempoolHistogramFeeRates& limits{feeLimits.value()};
+
+        /* Keep histogram per...
+         * ... cumulated tx sizes
+         * ... txns (count)
+         * ... cumulated fees */
+        std::vector<uint64_t> sizes(limits.size(), 0);
+        std::vector<uint64_t> count(limits.size(), 0);
+        std::vector<CAmount> fees(limits.size(), 0);
+
+        for (const CTxMemPoolEntry& e : pool.mapTx) {
+            const CAmount fee{e.GetFee()}, afees{e.GetModFeesWithAncestors()}, dfees{e.GetModFeesWithDescendants()};
+            const uint32_t size{(uint32_t)e.GetTxSize()}, asize{(uint32_t)e.GetSizeWithAncestors()}, dsize{(uint32_t)e.GetSizeWithDescendants()};
```

Comment:
Your question is probably related to Marco's comment here: https://github.com/bitcoin/bitcoin/pull/21422#discussion_r644739703. And as I said there, any guidance on this would be greatly appreciated.

Reply:
Why not use uint64_t for everything? `GetSizeWithAncestors()` returns a uint64_t too.

Reply:
Now we have:

```cpp
const CAmount fee{e.GetFee()};
const uint32_t size{(uint32_t)e.GetTxSize()};
const CAmount fee_rate{CFeeRate{fee, size}.GetFee(1)};
```

`CFeeRate` has the following constructor: `CFeeRate::CFeeRate(const CAmount& nFeePaid, uint32_t num_bytes)`. When I cast `size` to `uint64_t`, then I will get a warning unless I do some other casting.

Any suggestion how to improve this?